examples/experimental/openvino/yolo_v5/main.py

```diff
@@ -0,0 +1,151 @@
+"""
+ Copyright (c) 2022 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import os
+import subprocess
+from pathlib import Path
+
+import openvino.runtime as ov
+import torch
+import nncf
+
+from yolov5.utils.general import check_dataset
+from yolov5.utils.general import download
+from yolov5.utils.dataloaders import create_dataloader
+from yolov5.val import run as validation_fn
+
+
+FILE = Path(__file__).resolve()
+# Relative path to the `yolo_v5` directory.
+ROOT = Path(os.path.relpath(FILE.parent, Path.cwd()))
+# Path to the directory where the original and quantized IR will be saved.
+MODEL_DIR = ROOT.joinpath('yolov5m_quantization')
+# Path to the dataset config from the `ultralytics/yolov5` repository.
+DATASET_CONFIG = ROOT.joinpath('yolov5', 'data', 'coco.yaml')
+
+
+ie = ov.Core()
+
+
+def run_example():
+    """
+    Runs the YOLOv5 quantization example.
+    """
+    # Step 1: Initialize model from the PyTorch Hub.
+    repo_path = str(ROOT.joinpath('yolov5'))
+    model = torch.hub.load(repo_path, 'yolov5m', pretrained=True, source='local')
+    model.eval()
+
+    # Step 2: Converts PyTorch model to the OpenVINO model.
+    ov_model = convert_torch_to_openvino(model)
+
+    # Step 3: Create calibration dataset.
+    data_source = create_data_source()
+
+    # Step 4: Apply quantization algorithm.
+
+    # Define the transformation method. This method should take a data item returned
+    # per iteration through the `data_source` object and transform it into the model's
+    # expected input that can be used for the model inference.
+    def transform_fn(data_item):
+        images, *_ = data_item
+        images = images.float()
+        images = images / 255
+        images = images.cpu().detach().numpy()
+        return images
+
+    # Wrap framework-specific data source into the `nncf.Dataset` object.
+    calibration_dataet = nncf.Dataset(data_source, transform_fn)
+    quantized_model = nncf.quantize(ov_model, calibration_dataet, preset=nncf.QuantizationPreset.MIXED)
```

Comment:
Unless we plan this function to be available with one and the same signature for all backends, it should be kept in the `nncf.openvino` namespace.

Reply:
Thanks for the comment. We plan to do that. So this method should be under the `nncf` namespace.